### PR TITLE
feat: speedup provisioning of new nodes

### PR DESCRIPTION
**Pull request by jfroche**
Ping the nodeProvisioner as a new task enters the queue to speed up
provisioning of new NomadWorker

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java [hunk 1]
```diff
@@ -71,5 +83,40 @@
...
     }
-}
 
+    private static void fireOnStarted(final Cloud cloud, final Label label,
+            final Collection<NodeProvisioner.PlannedNode> plannedNodes) {
+        for (CloudProvisioningListener cl : CloudProvisioningListener.all()) {
+            try {
+                cl.onStarted(cloud, label, plannedNodes);
+            } catch (Error e) {
+                throw e;
```
Comment (multani) [inline src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java:93]: This is useless, isn't it?

File: src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java [hunk 2]
```diff
@@ -71,5 +83,40 @@
         LOGGER.log(Level.FINE, "Provisioning not complete, consulting remaining strategies");
         return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
     }
-}
 
+    private static void fireOnStarted(final Cloud cloud, final Label label,
```
Comment (multani) [inline src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java:87]: I'm not super familiar with the CloudProvisioningListener, but I would expect a comment like "force the onStarted event on the cloud provisioner in order to ..."

Do you think you can add a comment that would help a bit?

**Review by multani — CHANGES_REQUESTED** [review]
I tested this and it seems much faster indeed, thanks for the fix!

Can you have a look at the 2 comments below? Thanks again! :bow:

File: src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java (hunk 1 above)
Comment (jfroche) [reply src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java:93]: Did copy from: https://github.com/jenkinsci/jenkins/blob/e27b310065b3c036b5fc9d123f1d1d99d3058c00/core/src/main/java/hudson/slaves/NodeProvisioner.java#L904-L917

Also there have been discussion about propagating that exception further in here: https://github.com/jenkinsci/jenkins/pull/2591/files#r84593689

Tell me what you think, from here it seems to me that this catch needs to stay

Comment (multani) [reply src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java:93]: Thanks for the link, looks like we can keep this here then :+1:

File: src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java (hunk 2 above)
Comment (jfroche) [reply src/main/java/org/jenkinsci/plugins/nomad/NomadProvisioningStrategy.java:91]: Not sure my comment helps much, this is what I read - not sure who is using that event

**Pull request merged**